instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -121,18 +121,11 @@ def request_span_attributes(env:)
               'http.method' => env['REQUEST_METHOD'],
               'http.host' => env['HTTP_HOST'] || 'unknown',
               'http.scheme' => env['rack.url_scheme'],
-              'http.target' => fullpath(env)
+              'http.target' => env['QUERY_STRING'].empty? ? env['PATH_INFO'] : "#{env['PATH_INFO']}?#{env['QUERY_STRING']}"
```

Comment:
I have a suggestion to use functional style here but not sure if this is generating too many objects for our performance needs:
```suggestion
 [env['PATH_INFO'], env['QUERY_STRING']].join("?")
```
Is there a unit test that covers this functionality? 

Will `env['QUERY_STRING']` always return a `String` object? Will it ever return a `nil`?

Reply:
>I have a suggestion to use functional style here but not sure if this is generating too many objects for our performance needs

🤔 , definitely looks nicer.

> Is there a unit test that covers this functionality?

Hmm, I think it is missing a case for when a query string is present.  I'll add that in.

> Will env['QUERY_STRING'] always return a String object? Will it ever return a nil?

According to the spec it can be "empty" but is always required.  I think that means an empty string?
```
QUERY_STRING
The portion of the request URL that follows the ?, if any. May be empty, but is always required!
```

Reply:
Doesn't look like that suggestion works the way we want it to. 
irb(main):001:0> ["/", ""].join("?")
=> "/?"
irb(main):002:0> ["/", nil].join("?")
=> "/?"

We expect it to be `/`

Reply:
Ah fair because `QUERY_STRING` could be a blank string. Makes sense.